pkg/fqdn/name_manager_test.go

```diff
@@ -24,63 +26,62 @@ import (
 // add a rule, get same UUID label on repeat generations
 func (ds *FQDNTestSuite) TestNameManagerCIDRGeneration(c *C) {
 	var (
-		selIPMap map[api.FQDNSelector][]net.IP
+		selIPMap map[api.FQDNSelector][]netip.Addr
 
 		nameManager = NewNameManager(Config{
-			MinTTL: 1,
-			Cache:  NewDNSCache(0),
+			MinTTL:  1,
+			Cache:   NewDNSCache(0),
+			IPCache: &fakeIPCache{},
```

Comment:
nit: If you replace this with the MockIPCache, no need to declare a new one in this package.

The Upsert/Remove prefixes calls in the fake IPCache were also unused so far. Not sure if you intended to use those as a basis for some subsequent unit testing.

```
diff --git a/pkg/fqdn/name_manager_test.go b/pkg/fqdn/name_manager_test.go
index 7b9d6cd098f0..1273fba4580d 100644
--- a/pkg/fqdn/name_manager_test.go
+++ b/pkg/fqdn/name_manager_test.go
@@ -14,9 +14,8 @@ import (

        "github.com/cilium/cilium/pkg/checker"
        "github.com/cilium/cilium/pkg/fqdn/dns"
-       ipcacheTypes "github.com/cilium/cilium/pkg/ipcache/types"
        "github.com/cilium/cilium/pkg/policy/api"
-       "github.com/cilium/cilium/pkg/source"
+       testipcache "github.com/cilium/cilium/pkg/testutils/ipcache"
 )

 // TestNameManagerCIDRGeneration tests rule generation output:
@@ -31,7 +30,7 @@ func (ds *FQDNTestSuite) TestNameManagerCIDRGeneration(c *C) {
                nameManager = NewNameManager(Config{
                        MinTTL:  1,
                        Cache:   NewDNSCache(0),
-                       IPCache: &fakeIPCache{},
+                       IPCache: testipcache.NewMockIPCache(),

                        UpdateSelectors: func(ctx context.Context, selectorIPMapping map[api.FQDNSelector][]netip.Addr, _ <-chan struct{}) *sync.WaitGroup {
                                for k, v := range selectorIPMapping {
@@ -75,7 +74,7 @@ func (ds *FQDNTestSuite) TestNameManagerMultiIPUpdate(c *C) {
                nameManager = NewNameManager(Config{
                        MinTTL:  1,
                        Cache:   NewDNSCache(0),
-                       IPCache: &fakeIPCache{},
+                       IPCache: testipcache.NewMockIPCache(),

                        UpdateSelectors: func(ctx context.Context, selectorIPMapping map[api.FQDNSelector][]netip.Addr, _ <-chan struct{}) *sync.WaitGroup {
                                for k, v := range selectorIPMapping {
@@ -138,28 +137,3 @@ func (ds *FQDNTestSuite) TestNameManagerMultiIPUpdate(c *C) {
        nameManager.Unlock()

 }
-
-type fakeIPCache struct {
-       prefixes map[netip.Prefix]struct{}
-}
-
-func (i *fakeIPCache) UpsertPrefixes(prefixes []netip.Prefix, src source.Source, resource ipcacheTypes.ResourceID) <-chan struct{} {
-       if i.prefixes == nil {
-               i.prefixes = make(map[netip.Prefix]struct{})
-       }
-       for _, p := range prefixes {
-               i.prefixes[p] = struct{}{}
-       }
-       out := make(chan struct{})
-       close(out)
-       return out
-}
-
-func (i *fakeIPCache) RemovePrefixes(prefixes []netip.Prefix, src source.Source, resource ipcacheTypes.ResourceID) {
-       if i.prefixes == nil {
-               return
-       }
-       for _, p := range prefixes {
-               delete(i.prefixes, p)
-       }
-}
diff --git a/pkg/testutils/ipcache/ipcache.go b/pkg/testutils/ipcache/ipcache.go
index 89922bf5763f..3174794c1cb6 100644
--- a/pkg/testutils/ipcache/ipcache.go
+++ b/pkg/testutils/ipcache/ipcache.go
@@ -54,3 +54,12 @@ func (m *MockIPCache) DeleteOnMetadataMatch(IP string, source source.Source, nam
 func NewMockIPCache() *MockIPCache {
        return &MockIPCache{}
 }
+
+func (m *MockIPCache) UpsertPrefixes(prefixes []netip.Prefix, src source.Source, resource ipcacheTypes.ResourceID) <-chan struct{} {
+       out := make(chan struct{})
+       close(out)
+       return out
+}
+
+func (m *MockIPCache) RemovePrefixes(prefixes []netip.Prefix, src source.Source, resource ipcacheTypes.ResourceID) {
+}
```